src/sdc11073/provider/providerimpl.py

```diff
@@ -490,7 +485,10 @@ def stop_realtime_sample_loop(self):
             self.waveform_provider.stop()
 
     def get_xaddrs(self) -> list[str]:
-        addresses = self._wsdiscovery.get_active_addresses()  # these own IP addresses are currently used by discovery
+        if self._use_hostname:
+            addresses = [socket.gethostname()]
```

Comment:
from the documentation: [gethostname()](https://docs.python.org/3/library/socket.html#socket.gethostname) doesn’t always return the fully qualified domain name; use [getfqdn()](https://docs.python.org/3/library/socket.html#socket.getfqdn) for that.

Reply:
This PR is no branch from last master, it undoes some of the latest changes. 
A test is missing.

Reply:
fixed (wrong cut+paste)...